app/models/solid_queue/job/clearable.rb

```diff
@@ -11,7 +11,10 @@ module Clearable
 
       class_methods do
         def clear_finished_in_batches(batch_size: 500, finished_before: SolidQueue.clear_finished_jobs_after.ago)
-          clearable(finished_before: finished_before).in_batches(of: batch_size).delete_all
+          loop do
+            records_deleted = clearable(finished_before: finished_before).limit(batch_size).delete_all
```

Comment:
`finished_before` is stable: This ensures that we won't have infinite loops in fetching records.